.github/workflows/push.yml

```diff
@@ -57,6 +57,57 @@ jobs:
       - name: Publish test coverage
         uses: codecov/codecov-action@v4
 
+  tests-changes:
```

Comment:
I presume the intention behind this is to facilitate faster local development since now you can simply run `make testchanges`? For the push workflow we would still need to run everything since import dependencies are not enough to track all relevant tests.

Reply:
Could you add a comment here explaining the intention of running this as a workflow? Presumably to ensure the test changes section works correctly?

Reply:
The intention for this tool to be used for running both unit tests and integration tests on CI. It can save significant time, especially in this cases:
 - test only change; in that case 
 - change is local affects only set of tests (e.g. bundle tests).

In that case we won't need 2 workflows, only workflow via this tool will remain. Engineers will be able to opt out, e.g. by adding /fulltests in description or comment.



> For the push workflow we would still need to run everything since import dependencies are not enough to track all relevant tests.

Can you give counter-example? Note, it also establishes data dependencies (with some assumptions).